src/zkp/pisch.rs

```diff
@@ -6,8 +6,26 @@
 // License, Version 2.0 found in the LICENSE-APACHE file in the root directory
 // of this source tree.
 
-//! Implements the ZKP from Figure 22 of <https://eprint.iacr.org/2021/060.pdf>
-
+//! Implements a zero-knowledge proof of knowledge of discrete logarithm.
+//!
+//! More precisely, this module includes methods to create and verify a
+//! non-interactive zero-knowledge proof of knowledge of the discrete logarithm
+//! x of a group element X - x such that X = g^x for a known generator g. This
+//! is known as Schnorr's identification protocol.
+//!
+//! This implementation uses a standard Fiat-Shamir transformation to make the
+//! proof non-interactive. We only implement it for the group defined by the
+//! elliptic curve secp256k1. Although, this proof is a little different from
```

Comment:
todo: make `secp256k1` a link to either [the Bitcoin wiki page](https://en.bitcoin.it/wiki/Secp256k1) or [the k256 library](https://docs.rs/k256/latest/k256/).